### PR TITLE
WIP: Start on a stochastic gradient descent function

### PR DESCRIPTION
**Pull request by tomrtk**

### CODE REVIEW

File: Project2/csnet/optim.py [hunk 1]
```diff
@@ -1,2 +1,73 @@
...
+
+    X = np.column_stack((np.ones((x.shape[0], 1)), x))
+
+    for epoch in range(epochs):
+        if not silent:
+            print(f"Epoch {epoch}/{epochs}")
+
+        for x_sub, y_sub in _random_mini_batch_generator(X, y, batch_size):
+            theta = (
+                theta - lr * 2 * x_sub.T @ ((x_sub @ theta) - y_sub)
```
Comment (WilliamHo1999) [inline Project2/csnet/optim.py:55]: Hard coded to MSE.

We will need other cost functions (Such as binary cross entropy) later. 

However, we can change it later.
